sdk/scheduler/src/main/java/com/mesosphere/sdk/storage/Persister.java

```diff
@@ -4,41 +4,65 @@
 import java.util.Map;
 
 /**
- * This interface should be implemented in order to store and fetch data.
+ * A low-level interface for key/value storage in a tree structure.
+ *
+ * <p>Individual nodes may be the parent of other nodes. Some of these parent nodes may lack any data of their own. The
+ * root-level node (with path "", or "/") is considered to always be present.
+ *
+ * <p>This interface should be implemented in order to store and fetch data, with paths delimited by
+ * {@link com.mesosphere.sdk.storage.PersisterUtils#PATH_DELIM}.
  */
 public interface Persister {
+
```

Comment:
Reviewer note: This is just doing the following:
- renaming `delete()` to `deleteAll()` to be clearer on what the expected behavior is
- reordering the existing methods to be a bit easier to understand
- adding more thorough javadocs on behavior